### DISCUSSION TITLE
Fix the issue with rclone mounts or any fuse mounts when used as external storage

### DISCUSSION THREAD

**Discussion by SubhashBose** [nextcloud/all-in-one#2927] [Ideas] [answered]
### Is your feature request related to a problem? Please describe. 

Suppose I have mounted `/mnt/` as external mount by passing this flag during docker run `-e NEXTCLOUD_MOUNT="/mnt/"`

Then I use `rclone mount` to mount and remote storage within this directory, e.g, `/mnt/dropbox_rclone` so that I can access this mounted remote storage within Nextcloud. 

So far this works fine. But when I make any changes to this mounted path `/mnt/dropbox_rclone` form the host side, (adding/deleting file), then those changes are not reflected within Nextcloud container.

This issue is known to happen with any docker containers unless `bind-propagation=rshared` options are passed with `--mount type` flag 

I think this issue is applicable to any fuse filesystem mounts.

### Describe the solution you'd like

The solution is to pass on the mount flag to nextcloud container something like this
`--mount type=bind,source=/mnt,target=/mnt,bind-propagation=rshared`
(the important part is `bind-propagation=rshared`)

In regular docker containers, I can do this myself, but since nextcloud-aio-nextcloud docker container is not directly accessible to us this has to be implemented in the run script in mastercontainer.

I have tested with simple docker containers, after using the flag as above,  now all changes made by host to this rclone/fuse mounted path are immediately propagated within the docker container.

**Comment by Zoey2936** [comment]
see https://github.com/nextcloud/all-in-one/pull/2601

**Comment by SubhashBose** [comment]
Thank you for pointing this out to me.

If I understand the linked thread correctly, the issue is still unresolved due to some conflicting scenarios.

So, instead of one-size-fit-all solution, how about you provide us a mastercontainer flag, say `-e NEXTCLOUD_MOUNTFLAG=`, which will go directly to the --mount flag of nextcloud container, so that user can define the mount flag as per their requirements.  This will open up other possibilities as well.

**Comment by szaimen** [comment, marked as answer]
https://github.com/nextcloud/all-in-one/pull/2601#issuecomment-1629100996

**Reply by szaimen** [reply]
This is now released with v6.4.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by SubhashBose** [reply]
I am checking the final relaease 6.4.0, and i can confirm the mount now works as intended.. Above mentioned issue is fixed at least in filesystem level.

However, now the issue is with nextcloud which somehow not reflecting changes even though docker container filesystem can see the changes. Anyway, this is not related to docker, but related to nextcloud. It seems nextcloud stop scanning changed or slow when number of files in external mount is too large (which is my case, i need to check in detail).

Thanks for the docker fix.

**Reply by reply-nick** [reply]
after change #2601, I am getting this error:
Message: Server error: `POST http://localhost/v1.41/containers/nextcloud-aio-nextcloud/start` resulted in a `500 Internal Server Error` response: {&quot;message&quot;:&quot;path /mnt is mounted on / but it is not a shared mount&quot;}

my data folder is on a NFS share.
Are there any workarounds for this?
